.github/workflows/server-ci-master.yml

```diff
@@ -5,9 +5,27 @@ on:
       - master
       - cloud
       - release-*
-      - mono-repo*
 
 jobs:
   master-ci:
     uses: ./.github/workflows/server-ci-template.yml
     secrets: inherit
+  on-failure:
+    runs-on: ubuntu-latest
+    needs: master-ci
+    if: ${{ always() && needs.master-ci.result == 'failure' }}
```

Comment:
TIL that `always() && needs.master-ci.result == 'failure'` is not equivalent to just `needs.master-ci.result == 'failure'` :open_mouth:  That's very useful to know, thanks Jesse!